src/chainstate/stacks/boot/costs-2.clar

```diff
@@ -1 +1,567 @@
+        write_count: u0,
+        read_count: u1,
+        read_length: u1
+    })
+
+
+(define-read-only (cost_load_contract (n uint))
+    {
+        runtime: (linear n u1 u157),
+        write_length: u0,
+        write_count: u0,
+        ;; set to 3 because of the associated metadata loads
+        read_count: u3,
+        read_length: (linear n u1 u1)
+    })
+
+
+(define-read-only (cost_create_map (n uint))
+    {
+        runtime: (linear n u1 u1631),
+        write_length: (linear n u1 u1),
+        write_count: u1,
+        read_count: u0,
+        read_length: u0
+    })
+
+
+(define-read-only (cost_create_var (n uint))
+    {
+        runtime: (linear n u7 u2152),
+        write_length: (linear n u1 u1),
+        write_count: u2,
+        read_count: u0,
+        read_length: u0
+    })
+
+
+(define-read-only (cost_create_nft (n uint))
+    {
+        runtime: (linear n u1 u1610),
+        write_length: (linear n u1 u1),
+        write_count: u1,
+        read_count: u0,
+        read_length: u0
+    })
+
+
+(define-read-only (cost_create_ft (n uint))
+    {
+        runtime: u1972,
+        write_length: u1,
+        write_count: u2,
+        read_count: u0,
+        read_length: u0
+    })
+
+
+(define-read-only (cost_fetch_entry (n uint))
+    {
+        runtime: (linear n u1 u1539),
+        write_length: u0,
+        write_count: u0,
+        read_count: u1,
+        read_length: (linear n u1 u1)
+    })
+
+
+(define-read-only (cost_set_entry (n uint))
+    {
+        runtime: (linear n u4 u2204),
+        write_length: (linear n u1 u1),
+        write_count: u1,
+        read_count: u1,
+        read_length: u0
+    })
+
+
+(define-read-only (cost_fetch_var (n uint))
+    {
+        runtime: (linear n u1 u543),
+        write_length: u0,
+        write_count: u0,
+        read_count: u1,
+        read_length: (linear n u1 u1)
+    })
+
+
+(define-read-only (cost_set_var (n uint))
+    {
+        runtime: (linear n u5 u691),
+        write_length: (linear n u1 u1),
+        write_count: u1,
+        read_count: u1,
+        read_length: u0
+    })
+
+
+(define-read-only (cost_contract_storage (n uint))
+    {
+        runtime: (linear n u13 u7982),
+        write_length: (linear n u1 u1),
+        write_count: u1,
+        read_count: u0,
+        read_length: u0
+    })
+
+
+(define-read-only (cost_block_info (n uint))
+    {
+        runtime: u6321,
+        write_length: u0,
+        write_count: u0,
+        read_count: u1,
+        read_length: u1
+    })
+
+
+(define-read-only (cost_stx_balance (n uint))
+    {
+        runtime: u1385,
+        write_length: u0,
+        write_count: u0,
+        read_count: u1,
+        read_length: u1
+    })
+
+
+(define-read-only (cost_stx_transfer (n uint))
+    {
+        runtime: u1430,
+        write_length: u1,
+        write_count: u1,
+        read_count: u1,
+        read_length: u1
+    })
+
+
+(define-read-only (cost_ft_mint (n uint))
+    {
+        runtime: u1645,
```

Comment:
Interesting.  The higher runtime for minting implies that efficient contracts that don't care about `ft-get-supply` might just mint all the tokens to themselves, and then `ft-transfer?` them to owners instead of minting them on the fly.

Reply:
The benefit would be _very_ marginal -- these operations all incur 2 MARF reads and writes, each of which is 2/15k of the block limit (1.3*10^-4), whereas the savings in runtime is 1/5m (2*10^-7) -- it would have very little impact on the cost of the transaction to perform a transfer instead of a mint.